subgraphs/_reference_/src/prices/routers/UniswapForksRouter.ts

```diff
@@ -0,0 +1,200 @@
+  const usdcAddress = config.usdcAddress();
+
+  if (isLpToken(tokenAddress, ethAddress)) {
+    return getLpTokenPriceUsdc(tokenAddress);
+  }
+  return getPriceFromRouterUSDC(tokenAddress, usdcAddress);
+}
+
+export function getPriceFromRouterUSDC(
+  tokenAddress: Address,
+  usdcAddress: Address
+): CustomPriceType {
+  return getPriceFromRouter(tokenAddress, usdcAddress);
+}
+
+export function getPriceFromRouter(
+  token0Address: Address,
+  token1Address: Address
+): CustomPriceType {
+  const config = utils.getConfig();
+
+  const ethAddress = config.ethAddress();
+  const wethAddress = config.wethAddress();
+
+  // Construct swap path
+  const path: Address[] = [];
+  let numberOfJumps: BigInt;
+
+  // Convert ETH address to WETH
+  if (token0Address == ethAddress) token0Address = wethAddress;
+  if (token1Address == ethAddress) token1Address = wethAddress;
+
+  const inputTokenIsWeth: bool =
+    token0Address.equals(wethAddress) || token1Address.equals(wethAddress);
+
+  if (inputTokenIsWeth) {
+    // Path = [token0, weth] or [weth, token1]
+    numberOfJumps = BigInt.fromI32(1);
+
+    path.push(token0Address);
+    path.push(token1Address);
+  } else {
+    // Path = [token0, weth, token1]
+    numberOfJumps = BigInt.fromI32(2);
+
+    path.push(token0Address);
+    path.push(wethAddress);
+    path.push(token1Address);
+  }
+
+  const token0Decimals = utils.getTokenDecimals(token0Address);
+  const amountIn = constants.BIGINT_TEN.pow(token0Decimals.toI32() as u8);
+
+  const routerAddresses = config.uniswapForks();
+
+  let amountOut = constants.BIGINT_ZERO;
+  for (let idx = 0; idx < routerAddresses.length; idx++) {
+    const routerAddress = routerAddresses.at(idx);
+
+    const uniswapForkRouter = UniswapRouterContract.bind(routerAddress);
+    const amountOutArray = uniswapForkRouter.try_getAmountsOut(amountIn, path);
+
+    if (!amountOutArray.reverted) {
+      amountOut = amountOutArray.value[amountOutArray.value.length - 1];
+    }
+  }
+
+  const feeBips = BigInt.fromI32(30);
+
+  const amountOutBigDecimal = amountOut
+    .times(constants.BIGINT_TEN_THOUSAND)
+    .div(constants.BIGINT_TEN_THOUSAND.minus(feeBips.times(numberOfJumps)))
+    .toBigDecimal();
+
+  return CustomPriceType.initialize(
+    amountOutBigDecimal,
+    config.usdcTokenDecimals().toI32() as u8
+  );
+}
+
+export function getLpTokenPriceUsdc(tokenAddress: Address): CustomPriceType {
+  const uniSwapPair = UniswapPairContract.bind(tokenAddress);
+
+  const totalLiquidity: CustomPriceType =
+    getLpTokenTotalLiquidityUsdc(tokenAddress);
+  const totalSupply = utils.readValue<BigInt>(
+    uniSwapPair.try_totalSupply(),
+    constants.BIGINT_ZERO
+  );
+  if (totalSupply == constants.BIGINT_ZERO || totalLiquidity.reverted) {
+    return new CustomPriceType();
+  }
+
+  let pairDecimals: number;
+  const pairDecimalsCall = uniSwapPair.try_decimals();
+
+  if (pairDecimalsCall.reverted) {
+    pairDecimals = constants.DEFAULT_DECIMALS.toI32() as u8;
```

Comment:
Is it safe to default to some decimals if the call reverted? Would it be preferable to return 0 price instead of potentially returning wrong values silently?

Reply:
Seconding this, I have ran into issues with this before. If it is 0 thought it will also be wrong, maybe throwing an error and returning the empty `CustomPrice` type is best here, then when we run into situations like this in the future we can have a whitelist?

Reply:
Sounds Good. I am now returning empty `CustomPrice` if the pair decimals call is reverted.